Jetty 12.0.x documentation

gregpoulos opened the pull request:
This pull request includes two broad categories of change:

1. Editorial suggestions for the "Getting Started" section of the Operations Guide. These suggestions mostly revolve around things like fixing typos and tightening up the language.
2. Fixes for broken includes and references. These are largely related changes in naming conventions now that Jetty 12 features separate modules for EE8/9/10.

Commit:
fix broken include

Commit:
fix demo module example

Commit:
add documentation markup to fix documentation build error

Commit:
editorial suggestions for installation instructions

Commit:
editorial suggestions for quickstart guide

Commit:
fix unresolved directive errors

Commit:
fix file not found exception thrown during doc build

Commit:
editorial suggestions for deployment quickstart, plus some fixed references

lachlan-roberts commented:
@gregpoulos looks like you need to sign the Eclipse Contributor Agreement 
https://www.eclipse.org/legal/ecafaq.php

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/deploy.adoc:
```diff
@@ -41,12 +41,12 @@ mywebapp.war
 ====
 
-To deploy a standard web application, you need to enable the `deploy` module (see the `deploy` module complete definition xref:og-module-deploy[here]).
+To deploy a standard web application, we'll need to enable the xref:og-module-deploy[`deploy` module]:
```
The whole documentation is written using the second pronoun `you` and present tense (when possible), or passive voice (to avoid pronouns), so I would stick with `you need` rather than `we'll`.

gregpoulos replied:
Sounds good. Will do this here and for the following suggestions.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/deploy.adoc:
```diff
@@ -57,33 +57,34 @@ $ java -jar $JETTY_HOME/start.jar --add-module=deploy
 ----
 
-The `deploy` module creates the `$JETTY_BASE/webapps` directory, the directory where `+*.war+` files or web application directories should be copied so that Jetty can deploy them.
+Activating the `deploy` module will create the `$JETTY_BASE/webapps` directory, which is where Jetty will look for any `+*.war+` files or web application directories to deploy.
```
I prefer present tense, so "Activating ... creates", "where Jetty looks", etc.
Any particular reason to switch to future tense? 
Again I ask for consistency with the rest of the documentation.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/deploy.adoc:
```diff
@@ -57,33 +57,34 @@ $ java -jar $JETTY_HOME/start.jar --add-module=deploy
 [NOTE]
 ====
-The `deploy` module only provides the feature of deploying web applications.
+Activating Jetty's `deploy` module only enables web application deployment. Whether these web applications are served via clear-text HTTP/1.1, or secure HTTP/1.1, or secure HTTP/2, or HTTP/3 (or even all of these protocols) depends on whether the correspondent Jetty modules have been enabled.
```
See? Here you used the present tense "Activating ... enables" 😃 

Please use a new line for every sentence, since it makes it easier on revision and diffs, so "Whether these ..." should be on a new line (the whole documentation is written in this way).

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/deploy.adoc:
```diff
@@ -57,33 +57,34 @@ $ java -jar $JETTY_HOME/start.jar --add-module=deploy
 ====
 
-Now you need to copy a web application to the `$JETTY_BASE/webapps` directory, and you can use one of the demos shipped with Jetty:
+Now let's copy a web application to the `$JETTY_BASE/webapps` directory. We can use one of the demos shipped with Jetty:
```
Let's -> You (because Let's == let us, which is first person plural vs using second singular everywhere).
We -> You.
Second sentence on a new line.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/deploy.adoc:
```diff
@@ -57,33 +57,34 @@ $ java -jar $JETTY_HOME/start.jar --add-module=deploy
 
 ----
-$JETTY_BASE
+$ tree $JETTY_BASE
```
Remove this line, Windows users won't know what `tree` is.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/deploy.adoc:
```diff
@@ -57,33 +57,34 @@ $ java -jar $JETTY_HOME/start.jar --add-module=deploy
 ----
-$JETTY_BASE
+$ tree $JETTY_BASE
+/path/to/jetty.base
```
For consistency with other parts of documentation, either we change everywhere to `/path/to/jetty.base`, or we keep `$JETTY_BASE`.
I'd keep for now, eventually replace them all in a different PR.

gregpoulos replied:
I agree we should be consistent. Will make a note to tackle this in another PR.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/install.adoc:
```diff
@@ -14,16 +14,14 @@
-CAUTION: For Windows users, you should unpack Jetty to a path that does not contain spaces.
-
-The rest of the instructions in this documentation will refer to this location as `$JETTY_HOME`, or `${jetty.home}`.
+After the download, unpacking Eclipse Jetty will extract the files into a directory called `jetty-home-VERSION`, where `VERSION` is the version of Jetty that you downloaded. For example, installing Jetty {version} will create a directory called `jetty-home-{version}`.
```
New sentence on its own new line.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/install.adoc:
```diff
@@ -14,16 +14,14 @@
 IMPORTANT: It is important that *only* stable release versions are used in production environments.
 Versions that have been deprecated or are released as Milestones (M), Alpha, Beta or Release Candidates (RC) are *not* suitable for production as they may contain security flaws or incomplete/non-functioning feature sets.
 
+Unpack Eclipse Jetty file into a convenient location, such as `/opt`. The rest of the instructions in this documentation will refer to this location as `$JETTY_HOME`, or `${jetty.home}`.
```
New sentence on new line.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc:
```diff
@@ -14,18 +14,17 @@
-Recall from the xref:og-arch[architecture section] that Jetty is based on xref:og-modules[modules], that provides features, and on xref:og-arch-jetty-base[`$JETTY_BASE`], the place where you configure which module (and therefore which feature) you want to enable, and where you configure module parameters.
-
-Jetty is started by executing `$JETTY_HOME/start.jar` from within a `$JETTY_BASE` directory, so first we need to create a `$JETTY_BASE`:
+First we need to create a `$JETTY_BASE` directory.
```
We -> You.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc:
```diff
@@ -14,18 +14,17 @@
 ----
 
-If you try to start Jetty from an empty `$JETTY_BASE` you get:
+We'll start Jetty by executing `$JETTY_HOME/start.jar` from this directory. However, if we try to start Jetty from an empty `$JETTY_BASE`, it will complain that we haven't enabled any modules:
```
We -> You.
Present tense.
New sentence on new line.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc:
```diff
@@ -36,28 +35,32 @@ $ java -jar $JETTY_HOME/start.jar
 ----
 
-Jetty exited complaining that there are no modules enabled, since the `$JETTY_BASE` you just created is empty and therefore there is no configuration to read to assemble the Jetty server.
+Jetty uses a xref:og-modules[module] system to configure and assemble its web server; these modules are enabled and configured in xref:og-arch-jetty-base[`$JETTY_BASE`]. Since the `$JETTY_BASE` directory we just created is empty, Jetty has no configuration it can use to assemble the server.
```
```suggestion
Jetty uses a xref:og-modules[module system] to configure and assemble the server; these modules are enabled and configured in xref:og-arch-jetty-base[`$JETTY_BASE`].
Since the `$JETTY_BASE` directory you just created is empty, Jetty has no configuration it can use to assemble the server.
```

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc:
```diff
@@ -36,28 +35,32 @@ $ java -jar $JETTY_HOME/start.jar
 ----
 
-Let's try to enable the `http` module (see also xref:og-protocols-http[this section] for additional information):
+Let's try to enable the xref:og-protocols-http[`http`] module.
```
Let's -> You (it was wrong even before) 😅

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc:
```diff
@@ -68,24 +71,27 @@ $ java -jar $JETTY_HOME/start.jar
 ----
 
-Note how Jetty is listening on port `8080` for clear-text HTTP/1.1 connections.
+Jetty is listening on port `8080` for clear-text HTTP/1.1 connections. However, since it has no web applications deployed, it will just reply with `404 Not Found` to every request.
```
New sentence on new line.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc:
```diff
@@ -36,28 +35,32 @@ $ java -jar $JETTY_HOME/start.jar
-Now you can start Jetty:
+Note that Jetty has enabled a number of transitive dependencies (`server`, `logging-jetty`, and so on) in order to activate the `http` module.
+
+Now we can start Jetty:
```
We -> You.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc:
```diff
@@ -36,28 +35,32 @@ $ java -jar $JETTY_HOME/start.jar
 ----
 
-Now you can start Jetty:
+Note that Jetty has enabled a number of transitive dependencies (`server`, `logging-jetty`, and so on) in order to activate the `http` module.
```
```suggestion
When Jetty enables the `http` module, it also automatically enables a number of transitive dependencies of the `http` module, such as the `server` module, the `logging-jetty` module, and so on.
```

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc:
```diff
@@ -68,24 +71,27 @@ $ java -jar $JETTY_HOME/start.jar
+Jetty is listening on port `8080` for clear-text HTTP/1.1 connections. However, since it has no web applications deployed, it will just reply with `404 Not Found` to every request.
 
-After having enabled the `http` module, the `$JETTY_BASE` directory looks like this:
+Before we xref:og-begin-deploy[deploy our first web application], let's see what's happened to the `$JETTY_BASE` directory now that we've enabled the `http` module:
```
We/Our -> You/Your.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc:
```diff
@@ -68,24 +71,27 @@ $ java -jar $JETTY_HOME/start.jar
 [source,subs=verbatim]
 ----
-JETTY_BASE
+$ tree $JETTY_BASE
```
Remove.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc:
```diff
@@ -68,24 +71,27 @@ $ java -jar $JETTY_HOME/start.jar
 ----
-JETTY_BASE
+$ tree $JETTY_BASE
+/path/to/jetty.base
```
Same comment as previously.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc:
```diff
@@ -68,24 +71,27 @@ $ java -jar $JETTY_HOME/start.jar
-<1> The `resources/jetty-logging.properties` file has been created because the `http` modules depends on the `server` module, which in turn depends on the `logging` module; the `logging` module created this file that can be configured to control the server logging level.
-<2> The `start.d/` directory contains the configuration files for the modules.
-<3> The `start.d/http.ini` file is the `http` module configuration file, where you can specify values for the `http` module properties.
+<1> The `resources/jetty-logging.properties` file configures the server's logging level. This file was auto-generated when the `jetty-logging` module was activated as a transitive dependency of the `http` module.
```
Use semicolon instead of period to keep a single sentence.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc:
```diff
@@ -68,24 +71,27 @@ $ java -jar $JETTY_HOME/start.jar
-<2> The `start.d/` directory contains the configuration files for the modules.
-<3> The `start.d/http.ini` file is the `http` module configuration file, where you can specify values for the `http` module properties.
+<1> The `resources/jetty-logging.properties` file configures the server's logging level. This file was auto-generated when the `jetty-logging` module was activated as a transitive dependency of the `http` module.
+<2> The `start.d/` directory contains configuration files for any modules you have explicitly activated.
```
```suggestion
<2> The `start.d/` directory contains the `+*.ini+` configuration files for any modules you have explicitly activated.
```

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc:
```diff
@@ -68,24 +71,27 @@ $ java -jar $JETTY_HOME/start.jar
+<2> The `start.d/` directory contains configuration files for any modules you have explicitly activated.
+<3> The `start.d/http.ini` file is the `http` module configuration file, where you can specify values for the xref:og-module-http[`http` module properties].
+
+NOTE: By default, Jetty does not generate configuration files in `start.d/` for modules activated as transitive dependencies. To manually configure such a module, you should activate it directly via Jetty's `--add_module` flag.
```
```suggestion
[NOTE]
====
By default, Jetty does not generate `+*.ini+` configuration files in `start.d/` for modules activated as transitive dependencies. 
To manually configure such modules, you should activate them directly via Jetty's `--add-modules` flag.
====
```

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc:
```diff
@@ -68,24 +71,27 @@ $ java -jar $JETTY_HOME/start.jar
+NOTE: By default, Jetty does not generate configuration files in `start.d/` for modules activated as transitive dependencies. To manually configure such a module, you should activate it directly via Jetty's `--add_module` flag.
 
-In the `http.ini` file you can find the following content (among other content):
+In the `http.ini` file we can find the following (among other contents):
```
We -> You.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc:
```diff
@@ -96,15 +102,9 @@ In the `http.ini` file you can find the following content (among other content):
-----
-
-To make this change persistent, you can edit the `http.ini` file, uncomment the module property `jetty.http.port` and change its value to `9999`:
+Let's change that default port. Open `http.ini`, uncomment the line containing `jetty.http.port=`, and change its value to `9999`:
```
Let's -> You.
New sentence on new line.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc:
```diff
@@ -124,11 +124,15 @@ $ java -jar $JETTY_HOME/start.jar
 ----
 
-Note how Jetty is now listening on port `9999` for clear-text HTTP/1.1 connections.
+You can also specify the value of a module property when you start up Jetty. A property value specified on the command-line in this way will *override* the value configured in a module's `.ini` file.
```
New sentence on new line.
Use \`+*.ini+\` to indicate the `ini` files.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/jaspi/chapter.adoc:
```diff
@@ -27,29 +27,31 @@ Only modules conforming to the "Servlet Container Profile" with the ServerAuthMo
 
 ----
-include::{JETTY_HOME}/modules/jaspi.mod[]
+include::{JETTY_HOME}/modules/ee9-jaspi.mod[]
```
Let's always use ee10 for the examples.

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/jaspi/chapter.adoc:
```diff
@@ -27,29 +27,31 @@ Only modules conforming to the "Servlet Container Profile" with the ServerAuthMo
 
 ----
-$ java -jar $JETTY_HOME/start.jar --add-modules=jaspi
+$ java -jar $JETTY_HOME/start.jar --add-modules=ee9-jaspi
```
ee9 -> ee10

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/jaspi/chapter.adoc:
```diff
@@ -27,29 +27,31 @@ Only modules conforming to the "Servlet Container Profile" with the ServerAuthMo
-The `jaspi-demo` module illustrates setting up HTTP Basic Authentication using a Jakarta Authentication module that comes packaged with jetty: `org.eclipse.jetty.security.jaspi.modules.BasicAuthenticationAuthModule`, and applies it for a context named `/test`.
+The `ee9-jaspi-demo` and `ee10-jaspi-demo` modules illustrate setting up HTTP Basic Authentication using the EE9 and EE10 Jakarta Authentication modules that come packaged with Jetty.
+
+The following example uses Jetty's EE9 implementation of `AuthConfigProvider` to register a `ServerAuthModule` directly.
```
ee9 -> ee10

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/jaspi/chapter.adoc:
```diff
@@ -27,29 +27,31 @@ Only modules conforming to the "Servlet Container Profile" with the ServerAuthMo
 [source, xml]
 ----
-include::{JETTY_HOME}/etc/jaspi/jaspi-demo.xml[]
+include::{JETTY_HOME}/etc/jaspi/jetty-ee9-jaspi-demo.xml[]
```
ee9 -> ee10

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/jaspi/chapter.adoc:
```diff
@@ -61,8 +63,8 @@ The `CallerPrincipalCallback` and `GroupPrincipalCallback` do not require use of
 
 ----
-include::{JETTY_HOME}/modules/jaspi-default-auth-config-factory.mod[]
+include::{JETTY_HOME}/modules/ee9-jaspi-default-auth-config-factory.mod[]
```
ee9 -> ee10

sbordet replied:
@gregpoulos still using ee9 here? There is no such module for ee10?

gregpoulos replied:
Yikes, apparently I missed this one. It should be updated – I'll post a fix.

gregpoulos committed:
Update documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc

Co-authored-by: Simone Bordet <simone.bordet@gmail.com>

gregpoulos committed:
Update documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc

Co-authored-by: Simone Bordet <simone.bordet@gmail.com>

gregpoulos committed:
Update documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc

Co-authored-by: Simone Bordet <simone.bordet@gmail.com>

gregpoulos committed:
Update documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc

Co-authored-by: Simone Bordet <simone.bordet@gmail.com>

Commit:
Incorporate suggestions from sbordet

Commit:
Merge branch 'jetty-12.0.x-documentation' of https://github.com/gregpoulos/jetty.project into jetty-12.0.x-documentation

* 'jetty-12.0.x-documentation' of https://github.com/gregpoulos/jetty.project:
  Update documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc
  Update documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc
  Update documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc
  Update documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc

Commit:
Merge branch 'jetty-12.0.x-documentation' of https://github.com/gregpoulos/jetty.project into jetty-12.0.x-documentation

* 'jetty-12.0.x-documentation' of https://github.com/gregpoulos/jetty.project:
  Update documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc
  Update documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc
  Update documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc
  Update documentation/jetty-documentation/src/main/asciidoc/operations-guide/begin/start.adoc

Commit:
Merge branch 'jetty-12.0.x-documentation' of https://github.com/gregpoulos/jetty.project into jetty-12.0.x-documentation

* 'jetty-12.0.x-documentation' of https://github.com/gregpoulos/jetty.project:

gregpoulos commented:
I've incorporated all of @sbordet's suggestions.

I ran into an email mismatch issue regarding my ECA sign-off, since apparently the email I used for my Eclipse Foundation account wasn't the same one I use in my git settings. I just updated my Eclipse Foundation account and re-signed the ECA, so hopefully now I'm just waiting for a change to propagate through the system somewhere.

gregpoulos commented:
Update: Looks like the ECA sign-off situation has been resolved!

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/jaspi/chapter.adoc:
```diff
@@ -27,29 +27,31 @@ Only modules conforming to the "Servlet Container Profile" with the ServerAuthMo
 ===== Configure JASPI
 
-To enable the `jaspi` module you can use the following command (issued from within the `$JETTY_BASE` directory):
+Activate the `ee{9,10}-jaspi` module that matches your EE platform version.
```
If a module exists for ee9, then it should also exist for ee8, since we mechanically generate ee8 from ee9.
I'd say to always use `ee{8,9,10}` everywhere so here `ee{9,10}` -> `ee{8,9,10}`.

gregpoulos replied:
That's odd – when I run `java -jar $JETTY_HOME/start.jar --list-modules=* | grep jaspi` I get:
```
                            ee10-jaspi - Enables JASPI authentication for deployed web applications.
ee10-jaspi-default-auth-config-factory - Provides a DefaultAuthConfigFactory for jaspi
                       ee10-jaspi-demo - Enables JASPI basic authentication the /test context path.
                             ee9-jaspi - Enables JASPI authentication for deployed web applications.
 ee9-jaspi-default-auth-config-factory - Provides a DefaultAuthConfigFactory for jaspi
                        ee9-jaspi-demo - Enables JASPI basic authentication the /test context path.
```

sbordet replied:
@olamy do you know whether there is a `ee8-jaspi` module?

sbordet commented on documentation/jetty-documentation/src/main/asciidoc/operations-guide/jaspi/chapter.adoc:
```diff
@@ -61,8 +63,8 @@ The `CallerPrincipalCallback` and `GroupPrincipalCallback` do not require use of
 Jetty provides an implementation of the `AuthConfigFactory` interface which is used to register `AuthConfigProviders`. This can be replaced by a custom implementation by adding a custom module which provides `auth-config-factory`.
 This custom module must reference an XML file which sets a new instance of the `AuthConfigFactory` with the static method `AuthConfigFactory.setFactory()`.
-For an example of this see the `jaspi-default-auth-config-factory` module, which provides the default implementation used by Jetty.
+For an example of this see the `ee{9,10}-jaspi-default-auth-config-factory` module, which provides the default implementation used by Jetty.
```
`ee{9,10}` -> `ee{8,9,10}`.

Commit:
Use EE10 instead of EE9 in documentation examples
